test/test_common/test_time_system.h

```diff
@@ -61,15 +61,19 @@ class SingletonTimeSystemHelper {
 public:
   SingletonTimeSystemHelper() : time_system_(nullptr) {}
 
-  void set(TestTimeSystem* time_system) {
-    RELEASE_ASSERT(time_system_ == nullptr, "Unexpected reset of SingletonTimeSystemHelper");
-    time_system_.reset(time_system);
-  }
+  using MakeTimeSystemFn = std::function<std::unique_ptr<TestTimeSystem>()>;
 
-  TestTimeSystem* timeSystem() { return time_system_.get(); }
+  /**
+   * Returns a singleton time-system, creating a default one of there's not
+   * one already. This method is thread-safe.
+   *
+   * @return the thread system.
```

Comment:
nit: the time system (or just remove since obvious)

As long as you are here, another nit, can you make this function return a reference since it should never be null?

Reply:
done; that did have a small but non-zero blast-radius around initialization.